### ISSUE TITLE
HIGH Collections: edit page - upon save, Solr for Collection object should have appropriate edit_access and read_access

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#1543]
### Descriptive summary

When a collection is edited, given participants, and saved, update Solr document to include access defined in permissions_templates_access table for that collection.

### Rationale

The permissions_templates_access table is the canonical source identifying access for collections.

### Expected behavior

Solr field `edit_access_group_ssim=` permission_templates_agent field `agent_id` where `agent_type=='group' and access='manage'`.

Solr field `user_access_user_ssim=` permission_templates_agent field `agent_id` where `agent_type=='user' and access='view'`.

Solr field `read_access_group_ssim=` permission_templates_agent field `agent_id` where `agent_type=='group' and access='manage'`.

Solr field `read_access_user_ssim=` permission_templates_agent field `agent_id` where `agent_type=='user' and access='view'`.

### Steps to reproduce the behavior

1. Create 4 users:  me@ex.com, mgr@ex.com, dep@ex.com, vw@ex.com
1. Login as me@ex.com
1. Create a collection type
1. Create a collection of this type and save

Note the entries in the permissions_templates_access table that were created for the new collection.

1. Edit the collection
1. Add participants...
  1. assign mgr@ex.com as a manager
  1. assign dep@ex.com as a depositor
  1. assign vw@ex.com as a viewer

Verify that the solr document for the collection includes...

```
edit_access_group_ssim=['admins']
edit_access_user_ssim=['me@ex.com', 'mgr@ex.com']
view_access_user_ssim=['vw@ex.com']
```
Note: These may contain other groups/users if, for example, you set visibility=PUBLIC on the Discovery tab, which would set `view_access_group_ssim=['public'].

### Dependency

#1538 Collections: db tables - adjust permissions_templates to include source_type
#1523 Collections: routes - add routes needed for permissions_templates and permissions_templates_access for collection participants
#1540 Collections: new page - upon save, create entry for this collection in permission_templates
#1540 Collections: new page - upon save, add default access entries in permission_templates_access
#1378 Collections: Create sharing partial of the new/edit form
#1541 Collections: edit page - as participants are added, add the group/user with the given access role

### Related Work:

Dependencies plus...
#1544 Collections: edit page - upon save, Fedora for Collection object should have appropriate ACLs 
#1651 Collections: define methods for permissions service
#1545 Collections: index page - add processing that allows for a work to be created directly in this collection
#1427 Collections: admin show page (min: `Deposit new works through this collection` button)

**Comment by elrayle** [comment]
A couple of observations that differ from described issue...
* Participants are added/removed as the Add/Remove buttons are clicked.  It does not require clicking Update Collection button.  Solr doc is updated with each Add/Remove.
* Visibility is stored in the solr doc `read_group_access` (e.g. public=public, institutional=registered, private=_no_read_group_).  One of these is appended to participants access in solr doc as participants are added/removed.

Remaining Work:
* Noticed that solr doc access for admin set is not updated fully when admin set is created
* Need to double check what happens when visibility is updated.  Make sure updates to solr doc  for visibility also includes participants access in solr doc.

**Comment by elrayle** [comment]
Fixed by PR #1664
